libcloudforensics/gcp.py

```diff
@@ -91,6 +91,134 @@ def CreateService(service_name, api_version):
   return service
 
 
+class GoogleCloudLog:
+  """Class representing a Google Cloud Logs.
+
+  Attributes:
+    project_id: Project name.
+    gcl_api_client: Client to interact with GCP logging API.
+
+  Example use:
+    # pylint: disable=line-too-long
+    gcp = GoogleCloudLog(project_id='your_project_name')
+    gcp.ListLogs()
+    gcp.ExecuteQuery(filter='resource.type="gce_instance" labels."compute.googleapis.com/resource_name"="instance-1"')
+    See https://cloud.google.com/logging/docs/view/advanced-queries for filter details.
+  """
+
+  LOGGING_API_VERSION = 'v2'
+
+  def __init__(self, project_id):
+    """Initialize the GoogleCloudProject object.
+
+    Args:
+      project_id (str): The name of the project.
+    """
+
+    self.project_id = 'projects/' + project_id
+    self.gcl_api_client = None
+
+  def GclApi(self):
+    """Get a Google Compute Logging service object.
+
+    Returns:
+      apiclient.discovery.Resource: A Google Compute Logging service object.
+    """
+
+    if self.gcl_api_client:
+      return self.gcl_api_client
+    self.gcl_api_client = CreateService('logging', self.LOGGING_API_VERSION)
+    return self.gcl_api_client
+
+  def ListLogs(self):
+    """List logs in project.
+
+    Returns:
+      list: the project logs available.
+
+    Raises:
+      RuntimeError: If API call failed.
+    """
+
+    have_all_tokens = False
+    page_token = None
+    logs = []
+    while not have_all_tokens:
+      gcl_instance_client = self.GclApi().logs()
+      if page_token:
+        request = gcl_instance_client.list(
+            parent=self.project_id, pageToken=page_token)
+      else:
+        request = gcl_instance_client.list(parent=self.project_id)
+      try:
+        result = request.execute()
+      except (RefreshError, DefaultCredentialsError) as exception:
+        error_msg = (
+            '{0:s}\n'
+            'Something is wrong with your Application Default '
+            'Credentials. Try running: '
+            '$ gcloud auth application-default login'.format(str(exception)))
+        raise RuntimeError(error_msg)
+      if 'logNames' in result:
```

Comment:
This `if` statement can be removed by changing the `for` loop into `for logtypes in result.get('logNames', [])`